.github/workflows/ci.yml

```
@@ -1,32 +1,151 @@
+          fetch-depth: 0
+      
+      - name: Detect changes
+        id: detect
+        run: |
+          chmod +x .github/scripts/detect-changes.sh
+          .github/scripts/detect-changes.sh > changes.json
+          cat changes.json
+      
+      - name: Determine jobs to run
+        id: should_run
+        run: |
```

actionlint (1.7.8)

shellcheck reported issue in this script: SC2129:style:9:1: Consider using { cmd1; cmd2; } >> file instead of individual redirects

(shellcheck)